common/formats/src/format_csv.rs

```diff
@@ -215,51 +235,26 @@ impl InputFormat for CsvInputFormat {
         let mut state = std::mem::replace(state, self.create_state());
         let state = state.as_any().downcast_mut::<CsvInputState>().unwrap();
         let memory = std::mem::take(&mut state.memory);
+
         let memory_reader = MemoryReader::new(memory);
         let mut checkpoint_reader = NestedCheckpointReader::new(memory_reader);
 
         let mut row_index = 0;
         while !checkpoint_reader.eof()? {
-            for column_index in 0..deserializers.len() {
-                if checkpoint_reader.ignore_white_spaces_and_byte(self.field_delimiter)? {
-                    deserializers[column_index].de_default(&self.settings);
-                } else {
-                    deserializers[column_index]
-                        .de_text_csv(&mut checkpoint_reader, &self.settings)?;
-
-                    if column_index + 1 != deserializers.len() {
-                        checkpoint_reader
-                            .must_ignore_white_spaces_and_byte(self.field_delimiter)?;
-                    }
-                }
-            }
-
-            checkpoint_reader.ignore_white_spaces_and_byte(self.field_delimiter)?;
-
-            if let Some(delimiter) = &self.record_delimiter {
-                if !checkpoint_reader.ignore_white_spaces_and_byte(*delimiter)?
-                    && !checkpoint_reader.eof()?
-                {
-                    return Err(ErrorCode::BadBytes(format!(
-                        "Parse csv error at line {}",
-                        row_index
-                    )));
-                }
-            } else {
-                if (!checkpoint_reader.ignore_white_spaces_and_byte(b'\n')?
-                    && !checkpoint_reader.ignore_white_spaces_and_byte(b'\r')?)
-                    && !checkpoint_reader.eof()?
-                {
-                    return Err(ErrorCode::BadBytes(format!(
-                        "Parse csv error at line {}",
-                        row_index
-                    )));
-                }
-
-                // \r\n
-                checkpoint_reader.ignore_white_spaces_and_byte(b'\n')?;
+            checkpoint_reader.push_checkpoint();
```

Comment:
we should pop_checkpoint if read_row works fine.

Reply:
Yes, I forgot it.